.github/workflows/ci.yml

```diff
@@ -228,41 +228,36 @@ jobs:
           miniforge-variant: Mambaforge
           miniforge-version: latest
           use-mamba: true
-      - name: ASV installation
-        run: pip install asv==0.5.1
       - name: Running benchmarks
         run: |
           git remote add upstream https://github.com/modin-project/modin.git
           git fetch upstream
-          if git diff upstream/master --name-only | grep -q "^asv_bench/"; then
-              # ASV correctly creates environments for testing only from the branch
-              # with `master` name
-              git checkout -b master
+          if git diff upstream/main --name-only | grep -q "^asv_bench/"; then
               cd asv_bench
-              asv check -v
+
+              mamba env create -f ../environment-dev.yml
+              conda activate modin
+              pip install ..
 
               asv machine --yes
 
               # check Modin on Ray
-              asv run --quick --strict --show-stderr --launch-method=spawn \
+              asv run --quick --dry-run --python=same --strict --show-stderr --launch-method=spawn \
```

Comment:
For some reason the built-in ASV functionality for creating an environment does not work with the main branch. At the moment it seems easier to manually create environments.